### PR TITLE
CFINSPEC-167: Profile Signing Rollup

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

This PR
* stores the signed profile in binary format.
*  renamed the `inspec artifact` command to `inspec sign`
*  execute the signed profile using `inspec exec` command.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by Vasu1105** [comment]
@clintoncwolfe could you please review this?

File: lib/plugins/inspec-sign/lib/inspec-sign/base.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          end
+          puts "Successfully generated #{artifact_filename}"
+        end
+      end
+
+      def self.profile_verify(options)
+        file_to_verifiy = options["signed_profile"]
+        puts "Verifying #{file_to_verifiy}"
+
+        iaf_file = Inspec::IafFile.new(file_to_verifiy)
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-sign/lib/inspec-sign/base.rb]: file_to_verify, not verifiy

File: lib/plugins/inspec-sign/lib/inspec-sign/base.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if iaf_file.valid?
+          puts "Profile is valid."
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-sign/lib/inspec-sign/base.rb:100]: We should also exit with code 0

File: lib/plugins/inspec-sign/lib/inspec-sign/base.rb [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        else
+          puts "Profile is invalid"
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-sign/lib/inspec-sign/base.rb:105]: We should also exit with a distinctive code, likely a new exit code in Inspec::UI

File: lib/plugins/inspec-sign/lib/inspec-sign/base.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            raise "#{path_to_profile} doesn't appear to be a valid #{PRODUCT_NAME} profile"
+          end
+
+          yaml = YAML.load_file(p.to_s)
+          yaml = yaml.to_hash
+
+          unless yaml.key? "name"
+            raise "Profile is invalid, name is not defined"
+          end
+
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-sign/lib/inspec-sign/base.rb:127]: We may need to verify that the field profile_content_id is present and populated. We may need to inject it if it is missing.

File: test/functional/inspec_exec_signed_profile_test.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
+require "functional/helper"
+
+describe "inspec exec" do
+  include FunctionalHelper
+
+  def test_inspec_exec_signed_profile
+    prepare_examples do |dir|
+      skip_windows!
```
Comment (clintoncwolfe) [inline test/functional/inspec_exec_signed_profile_test.rb:9]: Why are we skipping windows at this point?

File: lib/plugins/inspec-sign/lib/inspec-sign/base.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+        open "#{options["keyname"]}.pem.pub", "w" do |io|
+          io.write key.public_key.to_pem
+        end
+      end
+
+      def self.profile_sign(options)
+        artifact = new
+        path_to_profile = options["profile"]
+
+        Dir.mktmpdir do |workdir|
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-sign/lib/inspec-sign/base.rb]: I don't think we can use a tempdir. This breaks when archive vendors a profile with local path dependencies.

File: test/functional/inspec_exec_signed_profile_test.rb (hunk 5 above)
Comment (Vasu1105) [reply test/functional/inspec_exec_signed_profile_test.rb:9]: On windows the temporary file path wherein the test keys are getting generated is not getting resolved. This has no issues on Windows when tested manually.

**Review by dkumaras — COMMENTED** [review]
@clintoncwolfe / @Vasu1105 - Pls check the suggested changes and confirm.

**Review by clintoncwolfe — APPROVED** [review]
A long road for this one, finally ready to merge!

**Pull request merged**